src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -1096,12 +1098,48 @@ internal void RemoveAdditionalDocument(IVisualStudioHostDocument document)
             lock (_gate)
             {
                 _additionalDocuments.Remove(document.Id);
-                _documentMonikers.Remove(document.Key.Moniker);
+                RemoveMoniker(document);
             }
 
             UninitializeAdditionalDocument(document);
         }
 
+        private void AddMoniker(IVisualStudioHostDocument document)
+        {
+            var moniker = document.Key.Moniker;
+            if (_documentMonikers.TryGetValue(moniker, out var value))
+            {
+                value.refCount++;
+                _documentMonikers[moniker] = (value.document, value.refCount);
```

Comment:
I must admit: using tuples here seems really terrifying and really likely to make a mistake....